edit: add ErrNoCommand

ktye opened the pull request:
Fixes #22 

I changed the name to ErrNoCommand. I think this is the standard naming convention.
There is an example in ktye/ui/cmd/sam I will update soon.

eaburns commented:
An `Err` prefix is for variables. An `Error` suffix is for types. https://github.com/golang/go/wiki/Errors

ktye committed:
edit: add NoCommandError

Edit returns NoCommandError if only an address is given.
The error message contains the new dot.
This allows searches without editing.

ktye commented:
You are right. I changed the name like you proposed. Is this ok for you?

eaburns commented:
I saw, thanks. However I'm traveling this week and I won't have time to
look at this change until after.

On Wed, Jan 30, 2019, 09:10 ktye <notifications@github.com wrote:

> You are right. I changed the name like you proposed. Is this ok for you?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/eaburns/T/pull/24#issuecomment-459026859>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAOXBxYqltJ2fGE0R8r6G5kS14wfONe-ks5vIdIJgaJpZM4aWS0l>
> .
>

eaburns commented:
The code looks good, but can you please add tests. I just have two in mind:
1) an empty command returns dot (which is what it says in the comment), and
2) some other address-only string returns that address.

Also, can you squash the changes into a single commit?

ktye commented:
How do I do this? If I push into the patch-1 branch of ktye/T isn't this enough?

eaburns commented:
You mean, how do you squash? Basically, I would like this whole change to be one commit in the history instead of 2 which it is now (and presumably 3 after tests). To do that, you can use the interactive rebaser:

First, do `git checkout -b backup; git checkout patch-1`. This will create a branch called "backup" that reflects the current state of "patch-1", in case things go poorly (as often happens in git).

Next, `git rebase -i master` will open an editor with one line showing each change made in your branch beyond the `master` branch. Each line will be preceded by a verb. The default is `pick`. This tells the rebaser what to do when you save+close the editor. Change all but the first occurrence of `pick` to `fixup`.

The file will look something like this:

```
pick e7966f edit: add ErrNoCommand
fixup 4782f rename ErrNoCommand to NoCommandError
fixup 12348 add tests for ErrNoCommand

# some comments down here
# added by git
# you can ignore them
```

Save and exit your editor.

The result should be that the branch has been re-written to be just like master, but with a single extra commit. The commit's message should be that of the `edit: add ErrNoCommand` commit, and the diffs should be that of all of your changes squashed together into a single commit.

You can push that to your branch using `git push --force`. When you do so, this pull request should update accordingly.

ktye commented:
Thank's for the explanation. I'll try.

Can I add `newdot` to testCases and add these tests to TestEdit?
```go
type testCase struct {
	edit   string
	newdot []int64
	want   string
	print  string // regex
	err    string // regex
}
```
```
{edit: "", err: "no command", newdot: []int64{0, 0}},
{edit: "N", err: "bad command N"},
{edit: "/l+/", err: "no command", newdot: []int64{2, 4}},
```
I would compare dots if newdot it not nil.

eaburns commented:
Sounds good. Thanks!

ktye commented:
is that gocyclo that is failing? edit_test.go has become too complicated for it.

ktye commented:
Is this now a single commit? I thought I did it right, but I'm not sure anymore.
Are you waiting for me to clean something up, or you had just no time to took at it yet?

eaburns commented:
Sorry, I didn't get a chance to look until now. It's still two commits.

ktye commented:
Now it should be fine. Besides the actual patch, it includes the test cases and the fix to gok.sh.

eaburns commented:
Looks good. Thanks.
